db/write_thread.cc

```diff
@@ -655,7 +649,25 @@ void WriteThread::ExitAsBatchGroupLeader(WriteGroup& write_group,
   }
 
   if (enable_pipelined_write_) {
-    // Notify writers don't write to memtable to exit.
+    // We insert a dummy Writer right before our current write_group. This
+    // allows us to unlink our write_group without the risk that a subsequent
+    // writer becomes a new leader and might overtake use before we can add our
```

Comment:
No, as long as our writer group is still in the list (i.e., reachable via `newest_writer`) there cannot be a new leader and therefore we cannot be overtaken. But suppose we have unlinked our group (let's call it A) and the writer list is now empty. Then a new writer can be added (let's call it B), which will become leader and therefore immediately perform its write. This thread can then continue to unlink its writer which results in a race to add the writers to the memtable writer queue. I.e., in the WAL writer queue we had [A, B] (in that order), while in the memtable writer queue we can end up with [B, A]. Not sure if this can be a problem, but the old code made sure this cannot happen, so I did the same. I changed the wording to make this more clear.